pkg/config/config.go

```diff
@@ -173,8 +174,38 @@ const (
 
 // MatchPattern matches a string against patterns defined in gardenctl config
 // If matched, the function creates and returns a PatternMatch from the provided target string
-func (config *Config) MatchPattern(value string) (*PatternMatch, error) {
-	for _, p := range config.MatchPatterns {
+func (config *Config) MatchPattern(value string, name string) (*PatternMatch, error) {
+	var patternMatch *PatternMatch
+
+	for _, g := range config.Gardens {
+		match, err := matchPattern(g.Patterns, value)
+		if err != nil {
+			return nil, err
+		}
+
+		if match != nil {
+			match.Garden = g.Name
+
+			if name == "" || g.Name == name {
+				// Directly return match of selected garden
+				return match, nil
+			}
+
+			patternMatch = match
+		}
+	}
+
+	if patternMatch != nil {
+		// Did not match pattern of current garden, but did match other pattern
+		return patternMatch, nil
+	}
+
+	return nil, errors.New("the provided value does not match any pattern")
+}
+
+// matchPattern matches pattern with provided list of patterns
```

Comment:
maybe it makes sense to add it in the go-doc that no error is returned in case the pattern did not match?

Reply:
✔️